### PR TITLE
prow-build-trusted: add kubernetes-external-secrets

### PR DESCRIPTION
**Pull request by spiffxp**
Followup to https://github.com/kubernetes/k8s.io/pull/2078

In support of https://github.com/kubernetes/test-infra/pull/22298 and https://github.com/kubernetes/test-infra/pull/22293

### CODE REVIEW

File: infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/kubernetes-external-secrets.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  labels:
+    app: kubernetes-external-secrets
+spec:
+  selector:
+    app: kubernetes-external-secrets
+  ports:
+    - protocol: TCP
+      port: 3001
+      name: prometheus
+      targetPort: prometheus
```
Comment (ameukam) [inline infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/kubernetes-external-secrets.yaml:16]: @spiffxp are you planning to have a monitoring stack in this cluster ?

File: infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/kubernetes-external-secrets.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      containers:
+        - name: kubernetes-external-secrets
+          image: "ghcr.io/external-secrets/kubernetes-external-secrets:7.0.1"
+          imagePullPolicy: Always
+          ports:
+          - name: prometheus
+            containerPort: 3001
+          env:
+          - name: "LOG_LEVEL"
+            value: "debug" # TODO(ameukam): switch to info
```
Comment (ameukam) [inline infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/kubernetes-external-secrets.yaml:45]: Comment can be removed. I think we should leave `debug` as the default log level.

File: infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/kubernetes-external-secrets.yaml (hunk 1 above)
Comment (spiffxp) [reply infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/kubernetes-external-secrets.yaml:16]: Regardless of whether there is a prometheus in this cluster, something should eventually scrape this. We've got metrics exposed by greenhouse and boskos in k8s-infra-prow-build.  Let me double-check whether I have this exposed similarly

**Comment by spiffxp** [comment]
I had to run `terraform apply` to get an external IP created, but the resources in here haven't yet been deployed.  Your call whether I do this pre-merge or post-merge, but I'll wait for approval before doing so.

File: infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/kubernetes-external-secrets.yaml (hunk 1 above)
Comment (spiffxp) [reply infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/kubernetes-external-secrets.yaml]: I add a `google_compute_address` in terraform and made this of type `LoadBalancerIP` like the other two prow services that currently get scraped by monitoring.k8s.io

**Comment by ameukam** [comment]
Let's do this post-merge. 

Also [pull-k8sio-terraform-prow-build-trusted](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/wg-k8s-infra/k8sio-presubmit.yaml#L79) was supposed to run but nothing happened. It's not blocking for this PR.

/lgtm
/hold
Remove hold when ready to deploy.

**Comment by spiffxp** [comment]
The run_if_changed regex has a typo:`prow-build-trusted/*.tf$` would match `prow-build-trusted/.tf` or `prow-build-trusted////.tf` but none of the files this PR changes.  If I need to make any changes in the jobs due to https://github.com/kubernetes/test-infra/pull/22463 I'll include as a fix

**Comment by spiffxp** [comment]
/hold
actually I want to wait for the postsubmits to change...

**Comment by spiffxp** [comment]
/hold cancel
ready to watch this deploy

**Comment by spiffxp** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-k8sio-deploy-prow-build-trusted-resources/1402077432638345216
First run failed with:
```
customresourcedefinition.apiextensions.k8s.io/externalsecrets.kubernetes-client.io created
namespace/kubernetes-external-secrets created
namespace/staging-test-pods unchanged
namespace/test-pods unchanged
Error from server (NotFound): error when creating "prow-build-trusted/resources/kubernetes-external-secrets.yaml": namespaces "kubernetes-external-secrets" not found
Error from server (NotFound): error when creating "prow-build-trusted/resources/kubernetes-external-secrets.yaml": namespaces "kubernetes-external-secrets" not found
```
Re-ran job, namespaces were found (need to redo job to create cluster-scoped resources before creating namespace-scoped resources)

But it also failed, with
```
Error from server (Forbidden): error when creating "prow-build-trusted/resources/kubernetes-external-secrets_rbac.yaml": clusterroles.rbac.authorization.k8s.io is forbidden: User "prow-deployer@k8s-infra-prow-build-trusted.iam.gserviceaccount.com" cannot create resource "clusterroles" in API group "rbac.authorization.k8s.io" at the cluster scope: requires one of ["container.clusterRoles.create"] permission(s).
Error from server (Forbidden): error when creating "prow-build-trusted/resources/kubernetes-external-secrets_rbac.yaml": clusterrolebindings.rbac.authorization.k8s.io is forbidden: User "prow-deployer@k8s-infra-prow-build-trusted.iam.gserviceaccount.com" cannot create resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope: requires one of ["container.clusterRoleBindings.create"] permission(s).
Error from server (Forbidden): error when creating "prow-build-trusted/resources/kubernetes-external-secrets_rbac.yaml": clusterrolebindings.rbac.authorization.k8s.io is forbidden: User "prow-deployer@k8s-infra-prow-build-trusted.iam.gserviceaccount.com" cannot create resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope: requires one of ["container.clusterRoleBindings.create"] permission(s).
```

**Comment by spiffxp** [comment]
It looks like there's no (non-service-agent) pre-defined role that contains `container.clusterRoles.create` except for `roles/container.admin`, so my options are to either:
1. manually deploy myself as a cluster admin
2. give prow-deployer container.admin
3. create a custom role with `container.clusterRoles.*` and `container.clusterRoleBindings.*` and give that to prow-deployer

I'm going to do 2 to confirm the job can deploy, then work on 3

**Comment by spiffxp** [comment]
```
gcloud projects add-iam-policy-binding \
  k8s-infra-prow-build-trusted \
  --member=serviceAccount:prow-deployer@k8s-infra-prow-build-trusted.iam.gserviceaccount.com \
  --role=roles/container.admin
```

**Comment by spiffxp** [comment]
Re-run succeeded: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-k8sio-deploy-prow-build-trusted-resources/1402087372769726464

Although, of note:
```
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
```

**Comment by spiffxp** [comment]
Opened https://github.com/kubernetes/k8s.io/pull/2156 for option 3

**Comment by ameukam** [comment]
> Re-run succeeded: [prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-k8sio-deploy-prow-build-trusted-resources/1402087372769726464](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-k8sio-deploy-prow-build-trusted-resources/1402087372769726464)
> 
> Although, of note:
> 
> ```
> Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
> ```

Fixed in the next major release : https://github.com/external-secrets/kubernetes-external-secrets/releases/tag/8.0.0. 

If you're not faster than me, I'll plan do the upgrade in the next weeks.

**Comment by spiffxp** [comment]
Now that https://github.com/kubernetes/k8s.io/pull/2156 has merged, I'm dropping the `roles/container.admin` binding
```
gcloud projects remove-iam-policy-binding \
  k8s-infra-prow-build-trusted \
  --member="serviceAccount:prow-deployer@k8s-infra-prow-build-trusted.iam.gserviceaccount.com" \
  --role="roles/container.admin"
```

**Pull request merged**